src/dataLoaders/notablePerson.ts

```diff
@@ -1,21 +1,22 @@
 import DataLoader from 'dataloader';
+import bluebird from 'bluebird';
 
 import { NotablePerson } from '../database/entities/NotablePerson';
-import { connection } from '../database/connection';
+import { SchemaContext } from '../typings/schemaContext';
 
-export const notablePersonBySlugLoader = new DataLoader<
-  string,
-  NotablePerson | undefined
->(async slugs => {
-  const db = await connection;
+export const createNotablePersonBySlugLoader = ({
+  connection,
+}: Pick<SchemaContext, 'connection'>) =>
+  new DataLoader<string, NotablePerson | undefined>(async slugs => {
+    return bluebird.map(
+      slugs,
+      async slug => {
+        if (slug) {
+          return connection.getRepository(NotablePerson).findOne({ slug });
+        }
 
-  return Promise.all(
-    slugs.map(async slug => {
-      if (slug) {
-        return db.getRepository(NotablePerson).findOne({ slug });
-      }
-
-      return undefined;
-    }),
-  );
-});
+        return undefined;
+      },
+      { concurrency: 3 },
```

Comment:
I guess because I'm not familiar with how things in this codebase work, it seems weird to me that the Notable Person loader expects an iterable of slugs, but the way you use the loader, such as in `resolvers/queries/notablePerons.ts`, you pass one slug at a time?

Also, why do we have a need to limit concurrency here?

Reply:
`dataloader`'s API expects any array so that it can batch (and maybe deduplicate?) the load operations.

> why do we have a need to limit concurrency

I thought `bluebird` limits this to 1 by default, although I couldn't find the default value documented. I guess it's infinity then? Not sure.

Reply:
If I'm reading [this](http://bluebirdjs.com/docs/api/promise.map.html) correctly, I think concurrency defaults to infinity?

```
Promise.map(
    Iterable<any>|Promise<Iterable<any>> input,
    function(any item, int index, int length) mapper,
    [Object {concurrency: int=Infinity} options]
) -> Promise
```

Reply:
Aha! I read the text but didn't check the type signature :sweat_smile: